### PR TITLE
Document controls

### PR DESCRIPTION
**Pull request by beroset**
This helps to address https://github.com/AsteroidOS/asteroid/issues/231

### CODE REVIEW

File: src/controls/qml/IconButton.qml [hunk 1]
```diff
@@ -19,6 +19,47 @@
...
+
+    Here is an example which displays a blue square centered in the screen and a button at
+    the bottom of the screen.  When the button is pressed, the square turns green.
+
+    \qml
+    import QtQuick 2.9
+    import org.asteroid.controls 1.0
+
+    Item {
+        anchors.centerIn: parent
```
Comment (FlorentRevest) [inline src/controls/qml/IconButton.qml:39]: This centerIn if redundant with the fill. Same comment for most examples

File: src/controls/qml/Label.qml [hunk 2]
```diff
@@ -18,6 +18,41 @@
...
+
+    \qml
+    import QtQuick 2.9
+    import org.asteroid.controls 1.0
+
+    Rectangle {
+        anchors.centerIn: parent
+        anchors.fill: parent
+        color: "yellow"
+        Rectangle {
```
Comment (FlorentRevest) [inline src/controls/qml/Label.qml:39]: It seems to me that we could keep the example more minimal by removing this rectangle ? We could just have the top level rectangle be blue for example

File: src/controls/qml/LabeledSwitch.qml [hunk 3]
```diff
@@ -20,6 +20,50 @@
 import QtQuick 2.9
 import org.asteroid.controls 1.0
 
+/*!
+    \qmltype LabeledSwitch
+    \inqmlmodule org.controls.asteroid 1.0
+
+    \brief This combines Label and Switch in a convenient package.
+
+    This is a LabeledSwitch.
```
Comment (FlorentRevest) [inline src/controls/qml/LabeledSwitch.qml:29]: Do we need this sentence ? I suppose it'd already be clear from the context ?

File: src/controls/qml/SpinnerDelegate.qml [hunk 4]
```diff
@@ -18,6 +18,49 @@
 import QtQuick 2.9
 import org.asteroid.controls 1.0
 
+/*!
+    \qmltype SpinnerDelegate
+    \inqmlmodule org.controls.asteroid 1.0
+
+    \brief Provides access to dimensions relative to a ratio of the screen width/height
+
+    This singleton provides methods for building a user interface that automatically scales based on
```
Comment (FlorentRevest) [inline src/controls/qml/SpinnerDelegate.qml:27]: This sentence is

**Review by FlorentRevest — COMMENTED** [review]
Cool addition! Thank you :) I like that most examples are similar such that once the reader has spent 10 seconds understanding one they also save 10 seconds of thinking for all the other ones ;)

**Comment by beroset** [comment]
The only components not yet documented at all are `TextBase` and `SegmentedArc` and there is a need to improve the `TextArea` documentation.  Reasons are:

1. TextBase - is it intended to be used or does it just serve as a base for TextField and TextArea?
2. TextArea - for some reason I was unable to build any test script which uses this.  It's under investigation now and may become a separate issue.
3. SegmentedArc - when I try to use this in a test page, I get "module "QtQuick.shapes" is not installed".  Not sure why yet.

File: src/controls/qml/IconButton.qml (hunk 1 above)
Comment (beroset) [reply src/controls/qml/IconButton.qml]: Fixed!

File: src/controls/qml/Label.qml (hunk 2 above)
Comment (beroset) [reply src/controls/qml/Label.qml:39]: Fixed

File: src/controls/qml/LabeledSwitch.qml (hunk 3 above)
Comment (beroset) [reply src/controls/qml/LabeledSwitch.qml]: Good point -- fixed.

File: src/controls/qml/SpinnerDelegate.qml (hunk 4 above)
Comment (beroset) [reply src/controls/qml/SpinnerDelegate.qml]: Now it is  
;)

**Comment by beroset** [comment]
During the documentation process, I found and fixed problems with both SegmentedArc and TestArea.

**Pull request merged**